internal/pkg/redis/streams/client.go

```diff
@@ -0,0 +1,200 @@
+	// Client used for functionality related to reading messages
+	subscribeClient RedisClient
+
+	// Client used for functionality related to sending messages
+	publishClient RedisClient
+}
+
+// NewClient creates a new Client based on the provided configuration.
+func NewClient(messageBusConfig types.MessageBusConfig) (Client, error) {
+	return NewClientWithCreator(messageBusConfig, NewGoRedisClientWrapper, tls.X509KeyPair, tls.LoadX509KeyPair)
+}
+
+// NewClientWithCreator creates a new Client based on the provided configuration while allowing more control on the
+// creation of the underlying entities such as certs, keys, and Redis clients
+func NewClientWithCreator(
+	messageBusConfig types.MessageBusConfig,
+	creator RedisClientCreator,
+	pairCreator pkg.X509KeyPairCreator,
+	keyLoader pkg.X509KeyLoader) (Client, error) {
+
+	// Parse Optional configuration properties
+	optionalClientConfiguration, err := NewClientConfiguration(messageBusConfig)
+	if err != nil {
+		return Client{}, err
+	}
+
+	// Parse TLS configuration properties
+	tlsConfigurationOptions := pkg.TlsConfigurationOptions{}
+	err = pkg.Load(messageBusConfig.Optional, &tlsConfigurationOptions)
+	if err != nil {
+		return Client{}, err
+	}
+
+	var publishClient, subscribeClient RedisClient
+
+	// Create underlying client to use when publishing
+	if !messageBusConfig.PublishHost.IsHostInfoEmpty() {
+		publishClient, err = createRedisClient(
+			messageBusConfig.PublishHost.GetHostURL(),
+			optionalClientConfiguration,
+			tlsConfigurationOptions,
+			creator,
+			pairCreator,
+			keyLoader)
+
+		if err != nil {
+			return Client{}, err
+		}
+	}
+
+	// Create underlying client to use when subscribing
+	if !messageBusConfig.SubscribeHost.IsHostInfoEmpty() {
+		subscribeClient, err = createRedisClient(
+			messageBusConfig.SubscribeHost.GetHostURL(),
+			optionalClientConfiguration,
+			tlsConfigurationOptions,
+			creator,
+			pairCreator,
+			keyLoader)
+
+		if err != nil {
+			return Client{}, err
+		}
+	}
+
+	return Client{
+		subscribeClient: subscribeClient,
+		publishClient:   publishClient,
+	}, nil
+}
+
+// Connect noop as preemptive connections are not needed.
+func (c Client) Connect() error {
+	// No need to connect, connection pooling is handled by the underlying client.
+	return nil
+}
+
+// Publish sends the provided message to appropriate Redis stream.
+func (c Client) Publish(message types.MessageEnvelope, topic string) error {
+	if c.publishClient == nil {
+		return pkg.NewMissingConfigurationErr("PublishHostInfo", "Unable to create a connection for publishing")
+	}
+
+	if topic == "" {
+		// Empty streams are not allowed for Redis
+		return pkg.NewInvalidTopicErr("", "Unable to publish to the invalid topic")
+	}
+	values := map[string]interface{}{
+		"CorrelationID": message.CorrelationID,
+		"Payload":       message.Payload,
+		"Checksum":      message.Checksum,
+		"ContentType":   message.ContentType,
+	}
+
+	return c.publishClient.AddToStream(topic, values)
```

Comment:
We have existing tech debt (mostly created by me) in that keys are not separated into their own namespace by simply using a character or two as a prefix, e.g. `gmm:foo` where _gmm:_ creates the namespace and _foo_ is the key. When we upgrade to Redis 6, we can start taking advantage of ACLs tied to keys that have namespace prefixes.

I suggest prepending `gmm:` to the topic.

Reply:
> * Please document use of Mockery. None of the other code uses it and there is nothing in the Makefile to regenerate the `RedisClient.go` file.
> * Please document the Redis information architecture being used. One of the challenges I've faced in working across the different repos is simply trying to figure out how the pieces come together.
> * Please add the copyright and license header to the files that are missing it

I will add work on adding this now. Ignore my previous re-request for a review. I will re-request once these issues are addressed.